frontend/src/components/Forms/CoffeeChatsForm/CoffeeChatsForm.tsx

```diff
@@ -0,0 +1,176 @@
+  const [slackLink, setSlackLink] = useState<string>('');
+
+  useEffect(() => {
+    MembersAPI.getAllMembers().then((members) => setMembersList(members));
+    CoffeeChatAPI.getCoffeeChatsByUser(userInfo).then((coffeeChat) => {
+      setApprovedChats(coffeeChat.filter((chat) => chat.status === 'approved'));
+      setPendingChats(coffeeChat.filter((chat) => chat.status === 'pending'));
+      setRejectedChats(coffeeChat.filter((chat) => chat.status === 'rejected'));
+      setIsChatLoading(false);
+    });
+  }, [userInfo]);
+
+  const coffeeChatExists = async (): Promise<boolean> => {
+    const chats = await CoffeeChatAPI.getCoffeeChatsByUser(userInfo);
+    return chats.some((chat) => chat.otherMember.netid === member?.netid);
+  };
+
+  const submitCoffeeChat = async () => {
+    if (!member) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Team Member Selected',
+        contentMsg: 'Please select a team member!'
+      });
+    } else if (!slackLink) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Image Entered',
+        contentMsg: 'Please enter an image link!'
+      });
+    } else if (!category) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Category Entered',
+        contentMsg: 'Please enter a category!'
+      });
+    } else if (member.netid === userInfo.netid) {
+      Emitters.generalError.emit({
+        headerMsg: 'Cannot Coffee Chat Yourself',
+        contentMsg: 'Please submit a coffee chat with another member!'
+      });
+    } else if (await coffeeChatExists()) {
+      Emitters.generalError.emit({
+        headerMsg: 'Coffee Chat Exists',
+        contentMsg: 'Please submit a coffee chat with a new member!'
+      });
+    } else {
+      const newCoffeeChat: CoffeeChat = {
+        uuid: '',
+        submitter: userInfo,
+        otherMember: member,
+        slackLink,
+        category,
+        status: 'pending' as Status,
+        date: new Date().getTime()
+      };
+
+      CoffeeChatAPI.createCoffeeChat(newCoffeeChat).then(() => {
+        setPendingChats((pending) => [...pending, newCoffeeChat]);
+      });
+      Emitters.generalSuccess.emit({
+        headerMsg: 'Coffee Chat submitted!',
+        contentMsg: `The leads were notified of your submission, and your coffee chat will be approved soon!`
+      });
+      setMember(undefined);
+      setCategory('');
+      setSlackLink('');
+    }
+  };
+
+  return (
+    <div>
+      <Form className={styles.form_style}>
+        <h1>Submit Coffee Chats</h1>
+        <p>
+          Submit your coffee chats to earn credit for the Coffee Chat Bingo! Fill out this form
+          after each chat to receive your credit.
+        </p>
+        <div className={styles.inline}>
+          <label className={styles.bold}>
+            Select a team member: <span className={styles.red_color}>*</span>
+          </label>
+          <div className={styles.center_and_flex}>
+            {membersList ? (
+              <Dropdown
```

Comment:
It appears right now that I can only search for people in the search bar by NetID. Ideally, we'd be able to search by netid and full name.